core/src/main/java/org/apache/brooklyn/core/objs/BrooklynObjectInternal.java

```diff
@@ -31,12 +31,12 @@
 
 import com.google.common.annotations.Beta;
 
-public interface BrooklynObjectInternal extends BrooklynObject, Rebindable {
+public interface BrooklynObjectInternal<PublicSelfType extends BrooklynObject, InternalSelfType extends BrooklynObjectInternal<PublicSelfType,InternalSelfType>> extends BrooklynObject, Rebindable {
```

Comment:
This interface is mostly used to cast an object to it and then use it. As such introducing generics here will make it awkward to use, having to suppress warnings (you already do `@SuppressWarnings({ "rawtypes", "unchecked" })` for each new usage). 

The generic types don't introduce value anyway, better follow the `getRebindSupport` pattern, where subclasses will override it with stronger typing.